crates/biome_formatter/src/comments.rs

```diff
@@ -1217,3 +1219,51 @@ pub fn is_doc_comment<L: Language>(comment: &SyntaxTriviaPieceComments<L>) -> bo
         }
     })
 }
+
+/// Returns `true` if `comment` is a documentation-style comment, specifically
+/// matching the JSDoc format where the comment:
+/// - spans over multiple lines
+/// - starts with two stars (like `/**`)
+///
+/// This is a special case of [self::is_alignable_comment].
+///
+/// # Examples
+///
+/// ```rs,ignore
+/// assert!(is_doc_comment(&parse_comment(r#"
+///     /**
+///      * Multiline doc comment
+///      */
+/// "#)));
+///
+/// // Non doc-comments
+/// assert!(!is_doc_comment(&parse_comment(r#"
+///     /*
+///      * Single star
+///      */
+/// "#)));
+///
+/// assert!(!is_doc_comment(&parse_comment(r#"/** has no line break */"#)));
+///
+/// assert!(!is_doc_comment(&parse_comment(r#"
+///     /**
+///      *
+///     this line doesn't start with a star
+///     */
+/// "#)));
+/// ```
+pub fn is_doc_comment<L: Language>(comment: &SyntaxTriviaPieceComments<L>) -> bool {
```

Comment:
This is now _truly_ checking for "doc" comments, as defined by JSDoc's syntax. Again, I think this should probably be in the `biome_js_formatter` crate rather than here, but all of the logic that uses it is in this generic one right now, too. Once we start implementing HTML formatting this will likely have to get moved, but that's a future time.